Add config for testing with Circle

gfontenot opened the pull request:
In order to test our setup, we want to generate a new project and then
run the tests of that project to make sure everything is compiling
properly. Ideally this should help us find compilation issues in the
template before they make it into master, and should also help make sure
we have our template setup process working correctly.

gfontenot committed:
Add config for testing with Circle

In order to test our setup, we want to generate a new project and then
run the tests of that project to make sure everything is compiling
properly. Ideally this should help us find compilation issues in the
template before they make it into master, and should also help make sure
we have our template setup process working correctly.

tonyd256 commented on bin/test:
```diff
@@ -0,0 +1,12 @@
+
+cd ./TestProject
+
+bin/test
```
this was at first confusing ... would it be worth renaming this file (not the on in the test project)?

gfontenot replied:
Do you have a name suggestion? I'm not sure that there's anything I can think of that would be less confusing.

sharplet replied:
Instead of renaming, what about this?

``` sh
(cd ./TestProject && bin/test)
```

I feel like that more clearly shows the intent, and resolves the naming ambiguity.

gfontenot replied:
Oh yep that works for me.

tonyd256 reviewed (approved):
Just on small comment

gfontenot committed:
Fail post gen hook if setup fails

gfontenot commented:
This is failing CI because Xcode isn't up to date on Circle. I'm like 90% sure that's true.

sharplet reviewed (approved):
Tests ok on my machine so I think it's fine to ignore Circle for now 👍 

What do you think about adding an exit trap to cleanup up the `TestProject` folder?

gfontenot committed:
Improve test script

 - Make it clearer that we're running the test project's tests
 - Clean up our test project when we're done

gfontenot committed:
Upgrade Carthage on CI

gfontenot commented:
CI is green 🎉
